share/initialization/common

```diff
@@ -182,7 +182,20 @@ fi
 : ${ECHO_STYLE:=RAW}
 
 # variables needed for command history
-HISTFILE=~/.yash_history HISTSIZE=5000
+
+if ! [ "${HISTFILE-}" ]; then
+    HISTFILE=${XDG_STATE_HOME:-~/.local/state}/yash/history
+
+    # handle old default history path
+    if [ -f ~/.yash_history ]; then
+        printf 'Note: The default history file location has been changed to "%s".\n' "$HISTFILE" >&2
```

Comment:
Should we use color escape codes to make these warnings more noticeable?

Reply:
Thanks, but I'm not very willing to colorize this kind of message because it might look like it's part of a command prompt.